app/controllers/api/service_plans_controller.rb

```diff
@@ -3,50 +3,36 @@
 class Api::ServicePlansController < Api::PlansBaseController
   before_action :authorize_service_plans!
 
-  before_action :activate_sidebar_menu
-
   activate_menu :serviceadmin, :subscriptions, :service_plans
   sublayout 'api/service'
 
-  def index
-    @new_plan = ServicePlan
-  end
+  helper_method :default_service_plan_data, :service_plans_table_data
+  delegate :default_service_plan_data, :service_plans_table_data, to: :presenter
 
-  def new
-    @plan = collection.build params[:service_plan]
-  end
-
-  def edit
-    @plan || raise(ActiveRecord::RecordNotFound)
-  end
+  alias service_plans plans
 
-  # class super metod which is Api::PlansBaseController#create
-  # to create plan same way as all plans
-  #
+  # rubocop:disable Lint/UselessMethodDefinition
   def create
-    super create_service_plan_params
+    super
```

Comment:
Maybe don't need these methods if you just delegate to `super`?

Reply:
As in `delegate :create, :update, :destroy, to: super`?

Reply:
I delegated it to super and it works in my local env. However, tests fail with the following error:
```
undefined method `super' for #<Api::ServicePlansController:0x000000001a0dff10>

app/controllers/api/service_plans_controller.rb:11:in `create'
lib/three_scale/middleware/multitenant.rb:113:in `_call'
lib/three_scale/middleware/multitenant.rb:108:in `call'
lib/gitlab/testing/request_inspector_middleware.rb:61:in `call'
lib/gitlab/testing/request_blocker_middleware.rb:73:in `call'
test/integration/admin/api/service_plans_controller_test.rb:142:in `block in <class:ProviderMemberTest>'
```

[integration tests](https://app.circleci.com/pipelines/github/3scale/porta/20605/workflows/4ed8c221-d6ae-4f26-8ea6-5d6a238a0a96/jobs/241421)

Reply:
Since all is green, I guess tests are fixed, right?

Reply:
Tests only fail when delegating these methods, that is:
```
delegate :create, :update, :destroy, to: super
```

So in the end I didn't do it, I couldn't make it work.